packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -675,6 +675,14 @@ module.exports = {
                   }).where;
               }
             }
+            
+            if (queryOpts.hasOwnProperty('query') &&
```

Comment:
Emm I don't think that it's related to this, It's probably related to the fact that we are not getting the right ids. the solution is here (https://github.com/strapi/strapi/pull/1833/files#diff-90e07d4e8a70f5625100fa0c449f7115R651) we need to handle the case of a non-dominant relation by just moving the `break` line instead of the IF block, so the code would be:
```
 case 'manyToMany': 
  if (association.dominant) {
    const arrayOfIds = (obj[association.alias] || []).map(related => {
      return related[ref.primaryKey] || related;
    });

    // Where.
    queryOpts.query = strapi.utils.models.convertParams(name, {
      // Construct the "where" query to only retrieve entries which are
      // related to this entry.
      [ref.primaryKey]: arrayOfIds,
      ...where.where
    }).where;
    break;
  }
  // falls through
```

Try it out and tell me if it works for you

Reply:
It looks like there's 2 separate issues here.

1) When querying a related model from the dominant model, and the related model has more than 1 result, then an empty array is always returned for the related model.

2) when querying a relation from a non dominant model the relation always returns null (even if there's only 1 result).

The first issue is solved by changing the operator symbol to ```IN```.

The second issue is only solved by removing the ```if (association.dominant) {``` conditional block completely from the ```manyToMany``` case statement. I tried your proposed solution of moving the break inside of the conditional but it didn't work.